packages/core/src/base/staticClient.ts

```diff
@@ -151,42 +124,40 @@ async function runFetch(
     options.queryParameters
   ).toString();
 
-  const fetchOptions: RequestInit = {
-    method: method
+  // Lets grab all the RequestInit defaults from the clientConfig
+  const defaultsFromClientConfig: RequestInit = {
+    cacheManager: options.client.clientConfig.cacheManager,
+    retry: options.client.clientConfig.retrySettings
   };
 
-  fetchOptions.headers = options.client.clientConfig.headers
-    ? _.clone(options.client.clientConfig.headers)
-    : {};
-
-  // make-fetch-happen sets connection header to keep-alive by default which
-  // keeps node running unless it is explicitly killed.
-  // If the user wants to keep the connection alive they can set the Connection
-  // header to 'keep-alive' and we'll respect it. Otherwise, we set it to "close".
-  const connectionHeader = getHeader("connection", fetchOptions.headers);
-  if (!fetchOptions.headers[connectionHeader]) {
-    fetchOptions.headers[connectionHeader] = "close";
-  }
+  // Let's create a request init object of all configurations in the current request
+  const currentOptionsFromRequest: RequestInit = {
+    method: method,
+    retry: options.retrySettings,
+    body: JSON.stringify(options.body)
+  };
 
-  // if headers have been given for just this call, merge those in
-  if (options.headers) {
-    _.merge(fetchOptions.headers, options.headers);
-  }
+  // Merging like this will copy items into a new object, this removes the need to clone and then merge as we were before.
+  let finalOptions = _.merge(
+    {},
+    defaultsFromClientConfig,
+    currentOptionsFromRequest
+  );
 
-  if (options.body) {
-    fetchOptions.body = JSON.stringify(options.body);
-    fetchOptions.headers["Content-Type"] = CONTENT_TYPE;
-  }
+  // Headers are treated separately to be able to move them into their own object.
+  const headers = new Headers(_.merge({}, options.client.clientConfig.headers));
 
-  // To disable response caching, set cacheManager to null
-  if (options.client.clientConfig.cacheManager) {
-    fetchOptions.cacheManager = options.client.clientConfig.cacheManager;
+  for (const [header, value] of Object.entries(options.headers || {})) {
+    headers.set(header, value);
   }
+  finalOptions["headers"] = headers;
```

Comment:
Why is this `["headers"]` and not `.headers`?

Reply:
Typescript didn't know what the type was in the IDE as it was merged from lodash.  This is so it won't complain about missing .headers during the assignment.